Aquality.Selenium.Core/src/Aquality.Selenium.Core/Visualization/DumpManager.cs

```diff
@@ -133,18 +135,43 @@ protected virtual DirectoryInfo CleanUpAndGetDumpDirectory(string dumpName = nul
             return dirInfo;
         }
 
+        protected virtual int GetMaxNameLengthOfDumpElements() => ElementsForVisualization.Max(element => element.Key == null ? 0 : element.Key.Length);
+
         protected virtual DirectoryInfo GetDumpDirectory(string dumpName = null)
         {
-            const int maxNameLenght = 40;
+            // get the maximum length of the name among the form elements for the dump
+            var maxNameLengthOfDumpElements = GetMaxNameLengthOfDumpElements() + ImageFormat.Length;
+
+            // get array of subfolders in dump name
+            string[] dumpSubfoldersNames = (dumpName ?? FormName).Split('\\');
+
+            // get invalid characters that can not be in folder name
             var invalid = new string(Path.GetInvalidFileNameChars()) + new string(Path.GetInvalidPathChars());
-            var name = dumpName ?? FormName;
-            foreach (var character in invalid)
+
+            // create new dump name without invalid chars for each subfolder
+            StringBuilder validDumpName = new StringBuilder();
+            foreach (string folderName in dumpSubfoldersNames)
+            {
+                string copy_folderName = folderName;
```

Comment:
please name variables in camelCase, and use 'var' for variables declaration, to follow codestyle of the project. Same above and below